### PR TITLE
Add custom error pages ( 404 and 500 )

### PR DESCRIPTION
**Pull request by NakedFool**
## What does this PR do and why?

Closes https://github.com/jbx-protocol/juice-interface/issues/2948 
Closes https://github.com/jbx-protocol/juice-interface/issues/2949

I've created custom error pages following this documentation - https://nextjs.org/docs/advanced-features/custom-error-page

Initially, I wanted to create folder `/errors` and include folders `/404` and `/500` inside but that is not possible currently, however now we have `/500` and `/404` folders inside `/pages` directory.

Let me know also what you think of the layout, I think no more text is needed as it is pretty clean, the image speaks for itself and we have a small paragraph that indicates what could be wrong.

One more thing, I guess we want to include translations here so I wrapped it with `LanguageProvider`, I do not think there is any outside magic that can do it differently, but I could be wrong!
## Screenshots or screen recordings

Both images are compressed from ~ 600kb to around ~200kb

image
image
## Acceptance checklist

- [x] I have evaluated the [Approval Guidelines](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#approval-guidelines) for this PR.
- [x] I have tested this PR in [all supported browsers](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#supported-browsers).
- [ ] I have tested this PR in dark mode and light mode (if applicable).

### CODE REVIEW

File: src/pages/404/index.page.tsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    <LanguageProvider>
+      <div className="flex flex-col items-center justify-center p-6 md:p-40">
+        <img
+          src={error404.src}
+          alt="Image"
+          className="mb-6 h-60 w-60 object-center md:h-80 md:w-80"
+        />
+        <Trans>
+          <p className="mb-6 text-sm text-grey-600 md:text-base">
+            This page could not be found.
```
Comment (tomquirk) [inline src/pages/404/index.page.tsx:22]: **suggestion** perhaps we should `Trans` this line and the link text separately, wdyt?

**Review by tomquirk — CHANGES_REQUESTED** [review]
@NakedFool yessss!!! Looks so great, nice work. Just one comment.

File: src/pages/404/index.page.tsx (hunk 1 above)
Comment (NakedFool) [reply src/pages/404/index.page.tsx:22]: gotcha, changing! I gotta ask what is the difference? asking out of curiosity so I know for future implementations 🍻 

Is it so the translation pot breaks it into the next line for translation so it does not appear as one message which it isn't?

Comment (tomquirk) [reply src/pages/404/index.page.tsx:22]: @NakedFool yes exactly! Whether to seperate the strings is a case-by-case decision. In this case, I could imagine us using `Go to homepage` text in multiple spots, so we may as well only translate that _once_ and use the same translation everywhere. It's also totally valid to wrap the whole thing in a `Trans` too. Developer discretion!

**Review by tomquirk — APPROVED** [review]

**Pull request merged**